### ISSUE TITLE
YUV422 generates corrupted file fails when AltiVec acceleration is enabled

### ISSUE THREAD

**Issue by madscientist159** [lu-zero/x264#9]
Output:
Almost completely corrupt file.  Enough data exists to start playback of digital artifacts and noise, but that is the extent of the output.

What works:
YUV420

Steps to reproduce:
Download Big Buck Bunny sample file from Wikipedia:
https://commons.wikimedia.org/wiki/File:Big_Buck_Bunny_first_23_seconds_1080p.ogv

Convert to yuv422 .y4m: 
```ffmpeg -i Big_Buck_Bunny_first_23_seconds_1080p.ogv -pix_fmt yuv422p file.y4m```

Note that the resultant y4m file plays fine with ffplay.

Try to encode with x264 on POWER9:
```x264 --threads 16 --fps 30 --keyint 30 --min-keyint 30 --scenecut 0 --preset ultrafast --tune grain --bitrate 25600 --output-csp i422 --trellis 0 -o Big_Buck_Bunny_encoding_failure.ogv file.y4m```

Note severe corruption on playback.

**Comment by madscientist159** [comment]
@lu-zero Is there anything I can to do assist with debugging?  This is blocking one of my pet projects on this end.

Thanks!

**Comment by madscientist159** [comment]
@lu-zero

**Comment by lu-zero** [comment]
I had yet another busy week, to check why it is broke the simplest thing is to disable all the accelerated codepaths and enable one by one and see what happens.

**Comment by madscientist159** [comment]
Traced this bug as far back as 145f3a6275802a649b8dedb49bb0e054caf31717 , it's still there.

Starting to wonder if this ever worked for yuv422 files.

**Comment by awilfox** [comment]
Not sure if this is at all helpful, but ffplay is showing a bunch of:

```
[h264 @ 0x1002dc38d40] mb_type 52 in P slice too large at 114 2/0   
[h264 @ 0x1002dc38d40] error while decoding MB 114 2
[h264 @ 0x1002dc38d40] concealing 7855 DC, 7855 AC, 7855 MV errors in P frame
```

```
[h264 @ 0x1002dc38d40] out of range intra chroma pred mode
[h264 @ 0x1002dc38d40] error while decoding MB 115 16
[h264 @ 0x1002dc38d40] concealing 6174 DC, 6174 AC, 6174 MV errors in P frame
Invalid UE golomb code=   0 aq=    0KB vq=13824KB sq=    0B f=0/0   
[h264 @ 0x1002d914ee0] mb_type -1094995534 in P slice too large at 119 17
[h264 @ 0x1002d914ee0] error while decoding MB 119 17
[h264 @ 0x1002d914ee0] concealing 6050 DC, 6050 AC, 6050 MV errors in P frame
[h264 @ 0x1002d930740] corrupted macroblock 90 0 (total_coeff=-1)   
[h264 @ 0x1002d930740] error while decoding MB 90 0
[h264 @ 0x1002d930740] concealing 8119 DC, 8119 AC, 8119 MV errors in P frame
```

**Comment by madscientist159** [comment]
I've traced the most severe breakage into x264_quant_2x2_dc_altivec().  Forcing x264 to use the C version of that function allows at least a recognizable picture to be shown.

I've also traced the remainder of the breakage down and will be posting a pull request shortly.  I need help with x264_quant_2x2_dc_altivec as I am not sure what is going wrong; the code dates from 2007 and appears to be highly optimized, but does **not** work with yuv422 streams.
